### PR TITLE
Using Polly for Blazor calls, and Reauthentication logic

### PR DESCRIPTION
**Pull request by dborgen2**
## What’s changing?

This PR achieves 3 things:

- Retries API calls for the following error codes using Polly: 408, 429, 500, 502, 503, and 504. More codes can be added easily in the future if necessary.
- Made the reauthentication process more user friendly by letting them know they need to be reauthenticated and will be redirected to EAuth.
- Upped the cookie timeout to 30 minutes (Security Control AC-12). Meaning the user can now go 30 minutes before being redirected to EAuth.

## Why?

Closes 1801 and 1839

## This PR has:

- [x] **JIRA ticket(s)** (_Must be listed in PR request and branch name should follow convention NAC-123-description_)<description>

- [x] **Security Related Changes** (_Security Tab questions have been answered on JIRA ticket(s)_)

- [x] **Commented source code** (_required on public classes and methods, and elsewhere as appropriate_)

- [ ] **Documented API changes** (_required on public classes and methods, and elsewhere as appropriate_)
    - [ ] External API documentation - OpenApi docs and Generated Docs (e.g. https://github.com/18F/piipan/tree/dev/docs/openapi) 
    - [ ] Internal API documentation (e.g. https://github.com/18F/piipan/tree/dev/metrics/docs/openapi)   

- [ ] **Developer documentation** (_for new build/test/API changes or complex portions of the system_)

- [x] **Automated unit tests** (_to maintain or increase level of code coverage_)

- [x] **Changes to IAC** (_add any deployment steps that require manual intervention to the draft release notes_)
    - [x] Have IaC scripts run successfully against an existing deployed environment?
    - [ ] Have IaC scripts run successfully against a newly deployed environment?
    - [ ] Have any new bash scripts been given the appropriate permissions? (e.g. git update-index --chmod=+x $script)
    - [ ] Do the IaC changes require deleting existing resources?

## Anything else?

_Add other details that are helpful for review here_

## Code Review Checklist:

- [ ] Automated checks all pass? (code builds successfully, tests pass, test coverage meets/exceeds threshold, etc.) 
- [ ] Have all requirements and Acceptance Criteria been met?
- **If there are code changes**
    - [ ] Is there no unnecessary complexity? Code is readable, maintainable, etc.
    - [ ] Do the changes follow SOLID principles?
    - [ ] Are the changes robust? Concurrency considered? Performance? Security (e.g. no sql injections)?
    - [ ] Are newly added dependencies needed and appropriate? Are dependency files (e.g. packages.json & packages.lock.json) updated?
    - [ ] For any new projects/solutions, have they been added to code coverage files?
    - [ ] Public classes & methods are documented    
- **If there are API changes**
    - [ ] Does API make sense? Useful and concise?
    - [ ] Are existing API interfaces/contracts intact? i.e. no breaking changes
    - [ ] Are API tests included?
    - [ ] Are API responses appropriate and documented? (200s, 201, 404s, etc.)
    - [ ] Are internals kept private and not leaked into the API?
    - [ ] External API documentation - OpenApi docs and Generated Docs 
    - [ ] Internal API documentation    
- **If there are IAC changes**
    - [ ] IAC Scripts are readable & maintainable
    - [ ] IAC Scripts are documented
    - [ ] IAC scripts pass required checks
    - [ ] IAC scripts were run successfully by author
- **General Logging Questions**
    - [ ] Do new capabilities (or changed capabilities) include logging?
    - [ ] Is PII not logged?
- **General Error Handling Questions**
    - [ ] Has exception handling been added for new capabilities (or changed capabilities)?
    - [ ] Are errors/exceptions properly communicated to end-users? Were there requirements for user facing error messages?
    - [ ] Are errors/exceptions logged?
    - [ ] Is PII kept out of error/exception logging?
- **If there are Developer Documentation Changes**
    - [ ] General Project documentation (under piipan\docs. e.g. iac.md, architecture.md, etc.)
    - [ ] Github documentation additions/changes (Readmes)
- **General automated test questions**
    - [ ] End-to-End Tests provided (if code implements new user story)
    - [ ] Are tests sufficient?
    - [ ] Corner cases considered?
    - [ ] Are performance tests needed?

### CODE REVIEW

File: query-tool/src/Piipan.QueryTool/Properties/ServiceDependencies/fns-app-querytool-dcb - Web Deploy/profile.arm.json [hunk 1]
```diff
@@ -0,0 +1,113 @@
+{
```
Comment (kcherukupalli) [inline query-tool/src/Piipan.QueryTool/Properties/ServiceDependencies/fns-app-querytool-dcb - Web Deploy/profile.arm.json:1]: Is this file accidentally checked in?

Comment (dborgen2) [reply query-tool/src/Piipan.QueryTool/Properties/ServiceDependencies/fns-app-querytool-dcb - Web Deploy/profile.arm.json:1]: Yes, good catch. Thanks, I'll remove it

File: components/src/Piipan.Components/Modals/RedirectToLoginModal.razor [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    }
+}
+
+
+<h2 class="usa-modal__heading" id="RedirectToLoginModal-heading">
+    Reauthentication Needed
+</h2>
+<div class="usa-prose">
+    <p id="RedirectToLoginModal-description">
+        Our system has determined that you need to be reauthenticated. Please click Continue to be redirected to EAuth.
```
Comment (bgfoshay) [inline components/src/Piipan.Components/Modals/RedirectToLoginModal.razor:37]: Question: Did Corey provide the text for this message? Or approve it? It makes sense to me. Just want to make sure we're good with it for users.

Comment (dborgen2) [reply components/src/Piipan.Components/Modals/RedirectToLoginModal.razor:37]: He didn't, but I did a quick demo with him and I've got him on this PR. He said was going to take a look at the wording and let me know if he wanted it switched.

Comment (bgfoshay) [reply components/src/Piipan.Components/Modals/RedirectToLoginModal.razor:37]: 👍 Sounds good. Just wanted to make sure he had the chance to review.

**Review by bgfoshay — APPROVED** [review]
👍
